### DISCUSSION TITLE
Hide PantryID from app users in src code

### DISCUSSION THREAD

**Discussion by mrdummy228** [imRohan/Pantry#154] [Q&A]
Hello! As i said, i am making my email service. Everything works, but i feel unsecure with knowing that people may just look at source code of my service. Well, it is fine, but they also can find out PantryID, and however use it for unwanted database changes (delete all users from db, get user password kv pairs, etc.). How can i hide pantryID in code so that users who look source code won't find it out?

**Comment by imRohan** [comment]
Great question! What's your application's stack look like?

My first suggestion would be to store and load the PantryID from an ENV variable if that's possible. That way your source code is free of secrets.

Treat the PantryID as an API key and apply best practices for your stack.

**Reply by mrdummy228** [reply]
If i will apply env variable, then what if others won't have it? It would give out an error or something, but it would definitely fail to connect to storage.
